logback-classic/src/main/java/ch/qos/logback/classic/pattern/ThreadIdConverter.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Logback: the reliable, generic, fast and flexible logging framework.
+ * Copyright (C) 1999-2015, QOS.ch. All rights reserved.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation
+ *
+ *   or (per the licensee's choosing)
+ *
+ * under the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation.
+ */
+package ch.qos.logback.classic.pattern;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+
+/**
+ * Returns the id of the current thread.
+ * 
+ * @author Roberto Cella
+ */
+public class ThreadIdConverter extends ClassicConverter {
+
+    public String convert(ILoggingEvent event) {
+        return String.valueOf(Thread.currentThread().getId());
```

Comment:
Should this be present in the `ILoggingEvent` and retrieved from there? Otherwise this can return the thread id of an internal logback thread, not the id of the actual caller thread, e.g. in an `AsyncAppender`. See `prepareForDeferredProcessing()` in the default impl class `LoggingEvent`.